socketio/src/asynchronous/client/builder.rs

```diff
@@ -343,26 +385,70 @@ impl ClientBuilder {
     /// }
     /// ```
     pub async fn connect(self) -> Result<Client> {
```

Comment:
Can be in a follow-up, since you're anyways doing multiple changes: Can this move to a seperate method in the Client? Similar to how we do it for sync transport: https://github.com/1c3t3a/rust-socketio/blob/main/socketio/src/client/client.rs#L212-L243.

Reply:
I agree. I found it odd that the thread was inside the builder instead of the client. Moving it to the client totally makes sense.